packages/react-button/package.json

```diff
@@ -63,5 +63,12 @@
       "minor",
       "patch"
     ]
+  },
+  "exports": {
+    ".": {
+      "types": "./lib/index.d.ts",
```

Comment:
Why is this needed? that is actually invalid javascript import, as if I'd import typescript declaration file browser/node would throw an error. can we remove this ?

Reply:
I'm following the `types` mention here in the node.js docs: 

https://nodejs.org/api/packages.html#community-conditions-definitions

Reply:
With javascript imports, this won't be resolved because it's not looking for "types". Conditionals allow the resolving system to use a string array of enabled conditionals to determine which path to take. When you spawn node, there's a `--conditions` flag you can use to indicate a flag. For example, if you were in a test environment, a package could export additional entries for that:

```
{
  test: {
     './test/': './lib/test/'
  }
}

This path would then only be importable if `--conditions test` were provided to node.

Reply:
Actually I've seen some packages include the `package.json` in the exports map so that consumers can import the definition file. They don't have to be purely javascript files. Technically you could have a `typescript` condition that lets you resolve imports to `/src/` paths.

Reply:
thanks for great explanation ! btw do you know if TypeScript will consume the `types` export map in future  - so we don't have duplicate definitions (`typings` - current way, `exports.typs` - the "new" way )

Reply:
It's resolving in nightlies but you need to use `nodenext` resolution.

Reply:
Looks like nodenext resolution is targeted for 4.7.